MementoMori/MementoMori.Server/Controllers/AuthController.cs

```diff
@@ -83,20 +84,69 @@ public async Task<ActionResult> Login([FromBody] LoginDetails loginDetails)
         [HttpGet("loginResponse")]
         public IActionResult GetLoginResponse()
         {
-            bool isLoggedIn = _authService.GetRequesterId(HttpContext).HasValue;
+            var userId = _authService.GetRequesterId(HttpContext);
+            if (!userId.HasValue)
+            {
+                return Ok(new LoginResponse { IsLoggedIn = false });
+            }
 
             var loginResponse = new LoginResponse
             {
-                IsLoggedIn = isLoggedIn
+                IsLoggedIn = true
             };
 
             return Ok(loginResponse);
         }
+
         [HttpPost("logout")]
         public IActionResult Logout()
         {
             _authService.RemoveCookie(HttpContext);
             return Ok();
         }
+
+        [HttpGet("color")]
+        public async Task<IActionResult> GetUserColor()
+        {
+            var userId = _authService.GetRequesterId(HttpContext);
+            if (!userId.HasValue)
+            {
+                return BadRequest(new { Message = "User ID not found." });
+            }
+
+            var user = await _authRepo.GetUserByIdAsync(userId.Value);
+            if (user == null)
+            {
+                return NotFound(new { Message = "User not found." });
+            }
+
+            return Ok(new { color = user.HeaderColor });
+        }
+
+        [HttpPost("newColor")]
+        public async Task<IActionResult> UpdateUserColor([FromBody] UpdateColorRequest request)
+        {
+            var userId = _authService.GetRequesterId(HttpContext);
+            if (!userId.HasValue)
+            {
+                return Unauthorized();
+            }
+
+            var user = await _authRepo.GetUserByIdAsync(userId.Value);
+            if (user == null)
+            {
+                return NotFound(new { Message = "User not found." });
+            }
+
+            user.HeaderColor = request.NewColor;
+
+            var success = await _authRepo.SaveAsync();
```

Comment:
If im not mistaken `success` = the number of lines changed in db, and if it actually fails it would throw an exception instead, and in that case it probably wont even continue executing the rest of the function, thus L144 wouldnt be hit.